config/initializers/location_search.rb

```diff
@@ -23,33 +24,26 @@ def lookup_type(type)
   end
 end
 
-DynamicSearch.class_eval do
-  attribute :location, GeoPoint
-  mapping _parent: { type: 'media_search' } do
-    indexes :location, type: 'geo_point'
-  end
-end
-
 Dynamic.class_eval do
-  def add_update_elasticsearch_dynamic_annotation_task_response_geolocation
-    return if self.get_field(:response_geolocation).nil?
+  def get_elasticsearch_options_dynamic_annotation_task_response_geolocation
+    return {} if self.get_field(:response_geolocation).nil?
     location = {}
     geojson = JSON.parse(self.get_field_value(:response_geolocation))
     coordinates = geojson['geometry']['coordinates']
     indexable = geojson['properties']['name']
 
     if coordinates[0] != 0 || coordinates[1] != 0
+      # re-compute long value before sending to Elasticsearch
       location = {
         lat: coordinates[0],
-        lon: coordinates[1]
+        lon: ((coordinates[1].to_f + 180) % 360) - 180
```

Comment:
what is this for?

Reply:
new versions of ES validate longitude value allowed range is  [-180..180]
I got an error `illegal longitude value`  while creating index from PG and used this formula to fix this error.